### ISSUE TITLE
Source releases doesn't contain source code of modules

### ISSUE THREAD

**Issue by jimon** [premake/premake-core#217]
modules/... folders are empty, which makes bootstrap script fail :(

**Comment by neo2buha** [comment]
have you tried git submodule init ?

**Comment by jimon** [comment]
I mean [v5.0.0-alpha5.zip](https://github.com/premake/premake-core/archive/v5.0.0-alpha5.zip) and [v5.0.0-alpha5.tar.gz](https://github.com/premake/premake-core/archive/v5.0.0-alpha5.tar.gz) from [here](https://github.com/premake/premake-core/releases). It's not relevant to git repository.

**Comment by neo2buha** [comment]
indeed @jimon ... submodules are not there
@starkos ... i ran the package action with master and source as params  and it seems to work fine ...
did you use this script ?
could it be that git submodule failed for some reason ?

or maybe when cleaning the .git folder os.rmdir removed the module folder as well ?
on what system did you run this ?

**Comment by starkos** [comment]
If you download the official source package `premake-5.0.0-alpha5-src.zip` everything is fine. This is the one that is built by the packaging script.

The problem is that GitHub adds two magical and very prominent "Source Code" entries to the bottom of the list, and they are broken. I don't create those, and I don't know how to get rid of them.

Any ideas? Thinking this is a GitHub bug to solve?

**Comment by starkos** [comment]
I've sent a support request to GitHub, and I've also added a note to the release to point people to the correct source code package. Hope that helps for now!

**Comment by akaStiX** [comment]
I've just downloaded **premake-5.0.0-alpha5-src.zip** and there are NO source code for modules...

**Comment by starkos** [comment]
I just downloaded **premake-5.0.0-alpha5-src.zip** and there ARE source code files for the modules. Where you getting it from? I tried from the download page and from the [release page](https://github.com/premake/premake-core/releases).

**Comment by akaStiX** [comment]
I am sorry, but I downloaded the **Source code (zip)**, which has slightly different name - **premake-5.0.0-alpha5-src.zip**

**Comment by jimon** [comment]
@starkos not sure what's going on :
1) I want to release page
2) downloaded premake-core-5.0.0-alpha5.zip
3) Opened it with 7-Zip archive tool (same result with if I use zip support in far manager)
4) Got this :
![1](https://cloud.githubusercontent.com/assets/1333661/9434709/c44f286a-4a41-11e5-8f48-e8dd11377b6c.png)

So looks like "there ARE source code files for the modules" it's not true at least for me.

**Comment by starkos** [comment]
The "**Source Code** (zip)" and "**Source Code** (tar.gz)" files are generated by GitHub, not by me. If you download either of those packages (as you are) you will not get the module scripts, the pre-generated project files, or the pre-embedded scripts.

If you download the `premake-5.0.0-alpha5-src.zip` you will end up with a folder named `premake-5.0.0-alpha`, **not** `premake-core-5.0.0-alpha`.

I've added a bold note to the end of the change log indicating the correct file to download. I have contacted GitHub and they are going to look into adding a feature to turn those files off or otherwise override them. And for the next alpha release I will look into committing the generated support files to the release branch so those extra downloads will work as expected.

I will leave this issue open as a reminder, and will close it once one of those fixes have been put into place.
